### ISSUE TITLE
Crash on starting reconstruction

### ISSUE THREAD

**Issue by LenaSailor** [colmap/colmap#156]
Story so far:

- Compiles without error on Linux 64 bit platform.
- GUI launches OK.
- Create db and load images OK. (set of 34 2000x1333 previously used successfully in Windows colmap and VisualSfM).
- Feature extraction OK.
- Feature matching OK. Can see them via db.
- Start reconstruction. Almost instant crash. GUI disappears.

Any ideas?

**Comment by ahojnnes** [comment]
Can you start from the command line with `./src/exe/colmap` and report the traceback you get on the command line?

**Comment by LenaSailor** [comment]
OK.  

The GUI starts up and this is reported:

homer@homer-XPS-8300 ~/Downloads/colmap-master/src/exe $ ./colmap
QApplication: invalid style override passed, ignoring it.

**Comment by ahojnnes** [comment]
Then what happens when you start the reconstruction? Also report the traceback on the command-line please.

**Comment by LenaSailor** [comment]
homer@homer-XPS-8300 ~/Downloads/colmap-master/src/exe $ ./colmap
QApplication: invalid style override passed, ignoring it.
*** Aborted at 1497540062 (unix time) try "date -d @1497540062" if you are using GNU date ***
PC: @           0x823c00 Eigen::internal::call_assignment_no_alias<>()
*** SIGSEGV (@0x0) received by PID 3714 (TID 0x7f8dc2ffd700) from PID 0; stack trace: ***
    @     0x7f8e06e25390 (unknown)
    @           0x823c00 Eigen::internal::call_assignment_no_alias<>()
    @           0xd310f1 ceres::internal::TrustRegionMinimizer::ComputeTrustRegionStep()
    @           0xd32970 ceres::internal::TrustRegionMinimizer::Minimize()
    @           0xd02093 ceres::Solver::Solve()
    @           0xd03009 ceres::Solve()
    @           0x84ab07 colmap::BundleAdjuster::Solve()
    @           0x781d4c colmap::IncrementalMapper::AdjustGlobalBundle()
    @           0x6c182f colmap::(anonymous namespace)::AdjustGlobalBundle()
    @           0x6c2006 colmap::IncrementalMapperController::Reconstruct()
    @           0x6c388b colmap::IncrementalMapperController::Run()
    @           0x77bddc colmap::Thread::RunFunc()
    @     0x7f8e03c3dc80 (unknown)
    @     0x7f8e06e1b6ba start_thread
    @     0x7f8e033a382d clone
    @                0x0 (unknown)
Segmentation fault

**Comment by ahojnnes** [comment]
This is a known issue under Ubuntu 16.04. You are using the default eigen version from the Ubuntu repository. Please, change to any other Eigen version. If you upgrade to the latest commits of Colmap on Github, you will see appropriate instructions and the build fails.

**Comment by ahojnnes** [comment]
Closing this issue for now, but feel free to reopen.

**Comment by LenaSailor** [comment]
Hi 

Following your suggestion of any other  eigen version (than the default) I tried 3.2.10.  Sadly that raised a Ceres issue.:

**homer@homer-XPS-8300 ~/Downloads/colmap-master/build $ cmake ../
-- Boost version: 1.58.0
-- Found the following Boost libraries:
--   program_options
--   filesystem
--   regex
--   system
--   unit_test_framework
-- Found Eigen3
--   Includes : /usr/include/eigen3
-- Found FreeImage
--   Includes : /usr/include
--   Libraries : /usr/lib/x86_64-linux-gnu/libfreeimage.so
-- Found Glog
--   Includes : /usr/include
--   Libraries : /usr/lib/x86_64-linux-gnu/libglog.so
-- Found installed version of Eigen: /usr/lib/cmake/eigen3
CMake Error at /usr/local/lib/cmake/Ceres/CeresConfig.cmake:88 (message):
  Failed to find Ceres - Missing required Ceres dependency: Eigen version
  3.2.92, please set EIGEN_INCLUDE_DIR.
Call Stack (most recent call first):
  /usr/local/lib/cmake/Ceres/CeresConfig.cmake:233 (ceres_report_not_found)
  CMakeLists.txt:76 (find_package)


CMake Error at CMakeLists.txt:76 (find_package):
  Found package configuration file:

    /usr/local/lib/cmake/Ceres/CeresConfig.cmake

  but it set Ceres_FOUND to FALSE so package "Ceres" is considered to be NOT
  FOUND.


-- Configuring incomplete, errors occurred!**

What would you suggest as best way to proceed?

Thanks

**Comment by ahojnnes** [comment]
As described in my error message, ceres must be compiled with the same Eigen version as Colmap. So you also have to recompile ceres. It's best to just clear your both build directories and recompile from scratch.

**Comment by LenaSailor** [comment]
Thanks for your prompt reply. I'll give it a shot.


⁣Sent from BlueMail ​

**Comment by insikk** [comment]
I  use Ubuntu 16.04, and got the same error. So I tried rebuild with Eigen3.3.4.

I donwloaded Eigen3.3.4 from http://eigen.tuxfamily.org/index.php?title=Main_Page#Documentation
unzip tar.gz file at `<PATH_TO_WORK_DIR>/eigen3.3.4`.
```
# Install Ceres 1.13.0 with Eigen 3.3.4
cd ceres
mkdir build
cd build
cmake .. -DEIGEN_INCLUDE_DIR=<PATH_TO_WORK_DIR>/eigen3.3.4/src
make -j8
make test
sudo make install

# Install colmap master branch commit 5a066b05234bec2cfe853604c7ca25718bb1afc7
# Author: Johannes Schönberger <jsch@demuc.de>
# Date:   Wed Nov 22 13:05:26 2017 -0500
# 
#     Add missing CMake copyright notice

cd colmap
mkdir build
cd build 
cmake .. -DEIGEN3_INCLUDE_DIRS=<PATH_TO_WORK_DIR>/eigen3.3.4/src
make -j8

# Run colmap 
./colmap 
QObject::~QObject: Timers cannot be stopped from another thread
QSocketNotifier: Invalid socket 31 and type 'Read', disabling...
*** Aborted at 1512356441 (unix time) try "date -d @1512356441" if you are using GNU date ***
PC: @     0x7f7b8f9c1da4 g_main_context_check
*** SIGSEGV (@0x7f7b965ebb68) received by PID 18261 (TID 0x7f7b966157c0) from PID 18446744071937375080; stack trace: ***
    @     0x7f7b958a2390 (unknown)
    @     0x7f7b8f9c1da4 g_main_context_check
    @     0x7f7b8f9c2330 (unknown)
    @     0x7f7b8f9c249c g_main_context_iteration

```

I am using south-building example, and trying to run automatic reconstruction with GUI app. 
Did I miss something?
Can anyone share how you did it in Ubuntu 16.04?

**Comment by insikk** [comment]
Following this issue(https://github.com/colmap/colmap/issues/87#issuecomment-294064080), I gave it try with Eigen3.3 Stable. 

And I also realized that, my error is not exactly the same as the previously posted error message. I stumbled upon [GPU timeout issue](https://colmap.github.io/faq.html#fix-gpu-freezes-and-timeouts-during-dense-reconstruction). And, Voila! I found out that my available VRAM is too small due to other process run some GPU computation.

After kill all processes using GPU, I can successfully run the south-building example with colmap GUI. It requires ~2.9GB VRAM for computation. 
![image](https://user-images.githubusercontent.com/2110359/33584264-ea1682a4-d9a1-11e7-90b8-7dd8d7be5c0b.png)
